src/api/app/controllers/source_package_command_controller.rb

```diff
@@ -140,23 +141,17 @@ def instantiate
 
   # POST /source/<project>/<package>?cmd=undelete
   def undelete
-    raise PackageExists, "the package exists already #{@target_project_name} #{@target_package_name}" if Package.exists_by_project_and_name(@target_project_name, @target_package_name, follow_project_links: false)
-
-    tprj = Project.get_by_name(@target_project_name)
-    raise CmdExecutionNoPermission, "no permission to create package in project #{@target_project_name}" unless tprj.is_a?(Project) && Pundit.policy(User.session, Package.new(project: tprj)).create?
-
-    path = request.path_info
+    raise PackageExists, "the package '#{@project.name}/#{params[:package]}' exists" if Package.exists_by_project_and_name(@project.name, params[:package], follow_project_links: false)
```

Comment:
Here, using `params[:package]` instead of `@target_package_name`, allows us to add the except: clause to the before filter for this action:
```ruby
before_action :set_target_package_name, except:  :undelete
```

Reply:
No it's still used in `authorize` and I want to move away from `@target_package_name` for `@project` and `@package`.